pkgs/applications/networking/mailreaders/tutanota-desktop/default.nix

```diff
@@ -1,61 +1,43 @@
-{ stdenv, lib, fetchurl, makeDesktopItem, copyDesktopItems, makeWrapper,
-electron, libsecret }:
+{ lib
+, appimageTools
+, fetchurl
+}:
 
-stdenv.mkDerivation rec {
+let
   pname = "tutanota-desktop";
-  version = "3.119.3";
+  version = "3.119.10";
 
   src = fetchurl {
-    url = "https://github.com/tutao/tutanota/releases/download/tutanota-desktop-release-${version}/${pname}-${version}-unpacked-linux.tar.gz";
-    name = "tutanota-desktop-${version}.tar.gz";
-    hash = "sha256-TdjvU12nh1sTfGTdBn+7dbEunaF38YjDvceEns4iRbA=";
+    url = "https://github.com/tutao/tutanota/releases/download/tutanota-desktop-release-${version}/tutanota-desktop-linux.AppImage";
+    hash = "sha256-yxj23SmCzgLqGAAgy7XzQYXcICFNbDBZQr8DJIK+oPs=";
   };
 
-  nativeBuildInputs = [
-    copyDesktopItems
-    makeWrapper
-  ];
-
-  dontConfigure = true;
-  dontBuild = true;
-
-  desktopItems = makeDesktopItem {
-    name = pname;
-    exec = "tutanota-desktop";
-    icon = "tutanota-desktop";
-    comment = meta.description;
-    desktopName = "Tutanota Desktop";
-    genericName = "Email Reader";
+  appimageContents = appimageTools.extractType2 {
```

Comment:
Use just `extract`

Reply:
Why use `extract` specifically? I ask this because `pkgs/by-name` has only two packages with `extract`:
```
grep -r 'appimageTools.extract' pkgs/by-name/
pkgs/by-name/hi/hifile/package.nix:  appimageContents = appimageTools.extractType2 {
pkgs/by-name/ca/caido/package.nix:  appimageContents = appimageTools.extractType2 { inherit pname src version; };
# HERE
pkgs/by-name/lu/lunar-client/package.nix:  appimageContents = appimageTools.extract { inherit pname version src; };
pkgs/by-name/be/beekeeper-studio/package.nix:  appimageContents = appimageTools.extractType2 { inherit pname version src; };
pkgs/by-name/an/anytype/package.nix:  appimageContents = appimageTools.extractType2 { inherit name src; };
pkgs/by-name/ar/arduino-ide/package.nix:  appimageContents = appimageTools.extractType2 { inherit pname version src; };
pkgs/by-name/li/listen1/package.nix:  appimageContents = appimageTools.extractType2 { inherit pname version src; };
# HERE
pkgs/by-name/si/simplex-chat-desktop/package.nix:  appimageContents = appimageTools.extract {
pkgs/by-name/sp/spacedrive/package.nix:      appimageContents = appimageTools.extractType2 { inherit pname version src; };
```
Is there something on the documentation specifying this?

Reply:
Documentation was updated recently, before that most maintainers just copied from each other outdated method. https://github.com/NixOS/nixpkgs/pull/276029

Reply:
Got it, that makes sense. Updated.